edgedb/protocol/codecs/namedtuple.pyx

```diff
@@ -70,15 +70,26 @@ cdef class NamedTupleCodec(BaseNamedRecordCodec):
         self._check_encoder()
 
         objlen = len(obj)
-        if objlen != len(self.fields_codecs):
+        fields_len = len(self.fields_codecs)
+
+        passed_arguments = set(obj.keys())
```

Comment:
This check is rather expensive so we only need to do it when we're certain that there's an error. So condition this with `if objlen <> fields_len:`

Reply:
I understand that, but I thought it would be nice to raise a `RuntimeError` with description here instead of a `KeyError` that might occur at the encoding in case of such query, for example:
```python3
conn.fetchall("""SELECT <int64>$arg_name""", wrong_name=42)
```
for this reason I replaced the original check. But restored it back.

Reply:
> I understand that, but I thought it would be nice to raise a RuntimeError with description here instead of a KeyError that might occur at the encoding in case of such query, for example:

I'm confused... even with my suggestion your latest PR code will raise a nice `RuntimeError`, right?

Reply:
Well, it will raise an error if there are either many or few arguments. But if the number of passed arguments is equal to what is required in the query, then `KeyError` will be raised there.

So these queries will raise a `RuntimeError`:

*  
  ```python3
  conn.fetchall("""SELECT <int64>$arg1""", arg1=1, arg2=2)  # extra {'arg2'}
  ```

* 
  ```python3
  conn.fetchall("""SELECT <int64>$arg1""", arg2=1, arg3=2)  # missed {'arg1'}, extra {'arg2', 'arg3'}
  ```

And this one will still raise a `KeyError`:

* 
  ```python3
  conn.fetchall("""SELECT <int64>$arg1""", arg2=2)  # KeyError: 'arg1'
  ```

And I'm not sure that such a difference would be good from the point of view of the user.

Maybe I should move the code that raises the error to a separate function (`raise_missmatched_args_error(fields_codecs, passed_args)` for example)? It will be called:
1. when the comparaision of the number of passed and required arguments fails.
2. when the value is encoded in case of `KeyError`.

Reply:
Got it. See my new comment that suggests how we can raise a nice error in all cases.